src/file-events/headers/linux_fsnotifier.h

```diff
@@ -0,0 +1,69 @@
+#pragma once
+
+#ifdef __linux__
+
+#include <poll.h>
+#include <sys/eventfd.h>
+#include <sys/inotify.h>
+#include <unordered_map>
+
+#include "generic_fsnotifier.h"
+#include "net_rubygrapefruit_platform_internal_jni_LinuxFileEventFunctions.h"
+
+using namespace std;
+
+class Server;
+
+struct Inotify {
+    Inotify();
+    ~Inotify();
+
+    const int fd;
```

Comment:
After this change it seems that `fd` should be private and the necessary methods should be added to `Inotify` so it can be. WYDT? As always, this can also be a follow up PR.

Reply:
I tried that before, but we still refer to it when polling. I don't see how we can easily make that go away. Note that the field is a `const`, therefore only read access is allowed to it, which is not as bad as if this was a mutable field.

Reply:
I think working with C++ and more importantly working with low-level C-style APIs allows two approaches: either model everything as an object, or accept that some primitives will be exposed. The extreme of the latter is to have everything as public `int`s, the extreme of the former is to reimplement the CLR. The current code seems to strike an acceptable balance for me.